assets/js/googlesitekit/datastore/user/feature-tours.js

```diff
@@ -0,0 +1,185 @@
+		return API.get( 'core', 'user', 'dismissed-tours', {}, { useCache: false } );
+	},
+	reducerCallback: ( state, dismissedTours ) => {
+		return {
+			...state,
+			dismissedTours,
+		};
+	},
+} );
+const { fetchGetDismissedTours, receiveGetDismissedTours } = fetchGetDismissedToursStore.actions;
+const fetchDismissTourStore = createFetchStore( {
+	baseName: 'dismissTour',
+	controlCallback: ( { slug } ) => API.set( 'core', 'user', 'dismiss-tour', { slug } ),
+	reducerCallback: ( state, dismissedTours ) => {
+		return {
+			...state,
+			dismissedTours,
+		};
+	},
+	argsToParams: ( slug ) => ( { slug } ),
+	validateParams: ( { slug } = {} ) => {
+		invariant( slug, 'slug is required.' );
+	},
+} );
+const { fetchDismissTour } = fetchDismissTourStore.actions;
+
+const baseInitialState = {
+	dismissedTours: undefined,
+};
+
+const baseActions = {
+	/**
+	 * Dismisses the given tour by slug.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} slug Tour slug to dismiss.
+	 * @return {Object} Generator instance.
+	 */
+	dismissTour( slug ) {
+		invariant( slug, 'A tour slug is required to dismiss a tour.' );
+
+		return ( function* () {
+			// Dismiss the given tour immediately.
+			yield {
+				payload: { slug },
+				type: DISMISS_TOUR,
+			};
+			// Dispatch a request to persist the dismissal.
+			const { response, error } = yield fetchDismissTour( slug );
+			if ( ! error ) {
+				yield receiveGetDismissedTours( response );
+			}
+			return { response, error };
+		}() );
+	},
+};
+
+const baseReducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case DISMISS_TOUR: {
+			const { slug } = payload;
+			const { dismissedTours = [] } = state;
+			if ( dismissedTours.includes( slug ) ) {
+				return state;
+			}
+			return {
+				...state,
+				dismissedTours: dismissedTours.concat( slug ),
+			};
+		}
+
+		default: {
+			return state;
+		}
+	}
+};
+
+const baseResolvers = {
+	*getDismissedTours() {
+		const { select } = yield Data.commonActions.getRegistry();
+		if ( ! select( STORE_NAME ).getDismissedTours() ) {
+			const { response, error } = yield fetchGetDismissedTours();
+			if ( ! error ) {
+				yield receiveGetDismissedTours( response );
+			}
+		}
+	},
+};
+
+const baseSelectors = {
+	/**
+	 * Gets the list of dismissed tours.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(string[]|undefined)} Array of dismissed tour slugs,
+	 *                                `undefined` if not resolved yet.
+	 */
+	getDismissedTours( state ) {
+		return state.dismissedTours;
+	},
+
+	/**
+	 * Checks whether or not the given tour is dismissed.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} slug  Tour slug to check.
+	 * @return {boolean} `true` if dismissed, otherwise `false`.
+	 */
+	isTourDismissed: createRegistrySelector( ( select ) => ( state, slug ) => {
+		const dismissedTours = select( STORE_NAME ).getDismissedTours() || [];
```

Comment:
Should we maybe return `true` if `getDismissedTours` hasn't resolved yet? When deciding whether a tour should be displayed, the current behavior could result the tour UI to briefly pop up before "recognizing" that it actually was already dismissed.

Reply:
Good point. Similar to other selectors, I don't think we should return `true` or `false` if the dependent selector hasn't been resolved yet. Let's return `undefined` if `getDismissedTours` is `undefined` so that this only returns a boolean if there is state to compute it, in which case it would always be accurate.